Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -228,63 +243,117 @@ open Chain
 
 variable [OmegaCompletePartialOrder β]
 variable [OmegaCompletePartialOrder γ]
+variable {f : α → β} {g : β → γ}
+
+/-- ωScottContinuous - Scott Continuous over Chains-/
+def ωScottContinuous (f : α → β) := ScottContinuousOn (Set.range fun c : Chain α => Set.range c) f
+
+lemma _root_.ScottContinuous.ωScottContinuous (hf : ScottContinuous f) : ωScottContinuous f :=
+  hf.scottContinuousOn
+
+lemma ωScottContinuous.monotone (h : ωScottContinuous f) : Monotone f :=
+  ScottContinuousOn.monotone _ (fun a b hab => by
+    use pair a b hab; exact range_pair a b hab) h
+
+lemma ωScottContinuous.isLUB {c : Chain α} (hf : ωScottContinuous f) :
+    IsLUB (Set.range (c.map ⟨f, hf.monotone⟩)) (f (ωSup c)) := by
+  simpa [map_coe, OrderHom.coe_mk, Set.range_comp]
+    using hf (by simp) (Set.range_nonempty _) (isChain_range c).directedOn (isLUB_range_ωSup c)
+
+lemma ωScottContinuous.id : ωScottContinuous (id : α → α) := ScottContinuousOn.id
+
+lemma ωScottContinuous.map_ωSup (hf : ωScottContinuous f) (c : Chain α) :
+    f (ωSup c) = ωSup (c.map ⟨f, hf.monotone⟩) := ωSup_eq_of_isLUB hf.isLUB
+
+lemma ωScottContinuous_iff_monotone_map_ωSup :
+    ωScottContinuous f ↔ ∃ hf : Monotone f, ∀ c : Chain α, f (ωSup c) = ωSup (c.map ⟨f, hf⟩) := by
+  refine ⟨fun hf ↦ ⟨hf.monotone, hf.map_ωSup⟩, ?_⟩
+  intro hf _ ⟨c, hc⟩ _ _ _ hda
+  convert isLUB_range_ωSup (c.map { toFun := f, monotone' := hf.1 })
+  · rw [map_coe, OrderHom.coe_mk, ← hc, ← (Set.range_comp f ⇑c)]
+  · rw [← hc] at hda
+    rw [← hf.2 c, ωSup_eq_of_isLUB hda]
+
+alias ⟨ωScottContinuous.monotone_map_ωSup, ωScottContinuous.of_monotone_map_ωSup⟩ :=
+  ωScottContinuous_iff_monotone_map_ωSup
+
+lemma ωScottContinuous.comp (hg : ωScottContinuous g) (hf : ωScottContinuous f) :
+    ωScottContinuous (g.comp f) :=
+  ωScottContinuous.of_monotone_map_ωSup
+    ⟨hg.monotone.comp hf.monotone, by simp [hf.map_ωSup, hg.map_ωSup, map_comp]⟩
+
+lemma ωScottContinuous.const {x : β} : ωScottContinuous (Function.const α x) := by
+  simp [ωScottContinuous, ScottContinuousOn, Set.range_nonempty]
+
+set_option linter.deprecated false
 
 /-- A monotone function `f : α →o β` is continuous if it distributes over ωSup.
```

Comment:
Let's preserve this docstring and similar below, probably by merging them into the new defs/lemmas

Reply:
Done, thanks.